### ISSUE TITLE
kinematics and dynamics

### ISSUE THREAD

**Issue by Lau-JW** [roboterax/humanoid-gym#33]
Where is the code for robot kinematics and dynamics? Additionally, I would like to migrate our project, PPO, to ROS2 Gazebo for simulation. Do you have any suggestions or references?

**Comment by zlw21gxy** [comment]
Because we are using RL only to control the robot, so we don't need to compute the kinematics and dynamics especially. But if you need to get the robot's state, like the Jacobian matrix or end effector position, you can use the Isaac Gym API. They can be obtained in the simulator. Also, what do you mean by PPO to Gazebo? Train PPO in the Gazebo environment with your own robot? Or infer the policy and render it with Gazebo?

**Comment by Lau-JW** [comment]
yes,i want to train PPO in the Gazebo environment with my robot . And i prefer ros2 simulation but don't know how to do.

**Comment by zlw21gxy** [comment]
Hi Coalwood,

Thank you for reaching out to me about your interest in training PPO in the
Gazebo environment with your robot.

I suggest reconsidering the use of Gazebo for rigid body simulation,
especially for legged robots. The sim-to-real gap can be quite significant,
which might hinder your results.

If your goal is to check your trained policy and integrate it with ROS 2, I
recommend modifying our sim2sim script. This way, you can utilize ROS 2 in
Python to pass messages and perform other necessary tasks.

If you have any further questions or need assistance with the setup, feel
free to ask!

**Comment by Lau-JW** [comment]
i import my robot and change default joint angles.
![截图 2024-10-12 16-01-07](https://github.com/user-attachments/assets/8d179df6-3e07-4bab-98ee-5ab91991a708)
but this occurred error, it appeared shoulders
![图片](https://github.com/user-attachments/assets/12bf6779-b2df-440b-9103-54dd54b3276b)
"self.dof_names[i]"  has  not only legs but also arm 
![图片](https://github.com/user-attachments/assets/17870492-112a-46f8-a044-b1b471ac5a80)
"self.dof_names[i]" is at this position,
![图片](https://github.com/user-attachments/assets/622735b1-ad14-4276-83f5-47347e1ba97d)

**Comment by Lau-JW** [comment]
![wx_camera_1728720636800.jpg](https://github.com/user-attachments/assets/efea48dc-b25a-40a1-a3cc-0777b33b0bc7)

**Comment by Lau-JW** [comment]
i import my robot and change default joint angles.
![截图 2024-10-12 16-01-07](https://github.com/user-attachments/assets/06f7beff-ba79-497d-a7b4-5faed80987d3)
but this occurred error, it appeared shoulders
![图片](https://github.com/user-attachments/assets/12bf6779-b2df-440b-9103-54dd54b3276b)
"self.dof_names[i]"  has  not only legs but also arm 
![截图 2024-10-12 16-03-00](https://github.com/user-attachments/assets/cf7611ed-694e-4a7e-81af-4191fef8a744)

![图片](https://github.com/user-attachments/assets/17870492-112a-46f8-a044-b1b471ac5a80)
"self.dof_names[i]" is at this position,
![图片](https://github.com/user-attachments/assets/622735b1-ad14-4276-83f5-47347e1ba97d)
